Sources/Apollo/ApolloInterceptorReentrantWrapper.swift

```diff
@@ -54,8 +54,6 @@ class ApolloInterceptorReentrantWrapper: RequestChain {
     if let cancellableInterceptor = wrappedInterceptor as? Cancellable {
       cancellableInterceptor.cancel()
     }
-
-    requestChain.takeUnretainedValue().cancel()
```

Comment:
This removal feels very wrong to me. Can you explain what is going on that makes you need to remove this?

Reply:
I removed it because it's cyclical. Note that this is in the reentrant wrapper so all calls to `cancel()` from all wrapped interceptors will be going back to the same 'parent' request chain but it will simply exit early because it's already been cancelled - [here](https://github.com/apollographql/apollo-ios/blob/feature/multipart-subscriptions/Sources/Apollo/InterceptorRequestChain.swift#L156). A couple lines down in that same function is where the call to cancel each cancellable wrapped interceptor is - [here](https://github.com/apollographql/apollo-ios/blob/feature/multipart-subscriptions/Sources/Apollo/InterceptorRequestChain.swift#L165).

All tests pass including the new retain cycles tests which are both on a query and a subscription, of which the subscription is required to be cancelled in order to release the unmanaged object.

Reply:
Is there any situation in which the `cancel()` function on the reentrant wrapper would be called directly, rather than via one of the wrapped interceptors? 

If it doesn't cause issues to have it, it feels like it's safer to keep it there for possible future iterations where you call `cancel()` directly here. But I may be thinking about this wrong. I don't have the context you have.

Reply:
The reentrant wrapper is [internal access](https://github.com/apollographql/apollo-ios/blob/feature/multipart-subscriptions/Sources/Apollo/ApolloInterceptorReentrantWrapper.swift#L9) only and the [list of wrapped interceptors on the request chain is private](https://github.com/apollographql/apollo-ios/blob/feature/multipart-subscriptions/Sources/Apollo/InterceptorRequestChain.swift#L23) so it's not possible publicly. We would have to be doing that internally which I feel goes against the design intent of it and _should_ break tests.

Reply:
Got it. Sounds good!